lib/kernel-manager.coffee

```diff
@@ -111,26 +112,23 @@ class KernelManager
         return kernelSpecs
 
 
-    getKernelSpecFor: (language) ->
+    getKernelSpecFor: (language, callback) ->
         unless language?
             return null
 
-        kernelMapping = Config.getJson('kernelMappings')?[language]
-        if kernelMapping?
-            kernelSpecs = @getAllKernelSpecs().filter (spec) ->
-                return spec.display_name is kernelMapping
+        kernelSpecs = @getAllKernelSpecsFor language
+        if kernelSpecs.length <= 1
+            callback kernelSpecs[0]
         else
-            kernelSpecs = @getAllKernelSpecsFor language
-
-        return kernelSpecs[0]
+            @kernelPicker = new KernelPicker ->
+                return kernelSpecs
+            @kernelPicker.onConfirmed = ({kernelSpec}) ->
+                callback kernelSpec
+            @kernelPicker.toggle()
 
 
     kernelSpecProvidesLanguage: (kernelSpec, language) ->
         kernelLanguage = kernelSpec.language
-        mappedLanguage = Config.getJson('languageMappings')[kernelLanguage]
```

Comment:
If `languageMappings` is removed, then users won't be able to use kernels with non-standard language names.